### PR TITLE
Bump dependencies.

### PR DESCRIPTION
**Pull request by kellertuer**
To ManifoldsBase.jl 0.15 and Manifolds.jl 0.9

### CODE REVIEW

File: ext/ManoptManifoldsExt/ARC_CG.jl [hunk 1]
```diff
@@ -1,8 +1,8 @@
-function set_manopt_parameter!(M::TangentSpaceAtPoint, ::Val{:p}, v)
+function set_manopt_parameter!(M::TangentSpace, ::Val{:p}, v)
```
Comment (mateuszbaran) [inline ext/ManoptManifoldsExt/ARC_CG.jl:1]: I think we could even move this from extension to core Manopt.

Comment (kellertuer) [reply ext/ManoptManifoldsExt/ARC_CG.jl:1]: You are right, nice! Will do that later today (first have a lecture now soon).

File: ext/ManoptManifoldsExt/ARC_CG.jl [hunk 2]
```diff
@@ -14,19 +14,19 @@ function (f::Manopt.AdaptiveRegularizationCubicCost)(M::TangentSpaceAtPoint, X)
...
+function (grad_f::Manopt.AdaptiveRegularizationCubicGrad)(M::TangentSpace, Y, X)
     get_hessian!(base_manifold(M), Y, grad_f.mho, M.point, X)
     Y .= Y + grad_f.X + grad_f.σ * norm(base_manifold(M), M.point, X) * X
     return Y
 end
 function (c::StopWhenFirstOrderProgress)(
-    dmp::AbstractManoptProblem{<:TangentSpaceAtPoint},
+    dmp::AbstractManoptProblem{<:TangentSpace},
     ams::AbstractManoptSolverState,
     i::Int,
```
[the remark this answers is not in the document]
Comment (kellertuer) [reply ext/ManoptManifoldsExt/ARC_CG.jl:31]: I am not sure how this happened I did run formatter. But well. CIs work in mysterious ways.

File: ext/ManoptManifoldsExt/ARC_CG.jl (hunk 1 above)
Comment (mateuszbaran) [reply ext/ManoptManifoldsExt/ARC_CG.jl:1]: Sure. `alternating_gradient.jl` from extension should also work in core Manopt now.

**Comment by kellertuer** [comment]
Hm, this might be more complicated than I thought, since RecursiveArrays are a dependency of Manifolds but only a weak dependency of ManifoldsBAse, so the alternating gradient can actually not be moved out I think – and if so only into another weak dependency with RecursiveArrays since I need the ArrayPartition?

**Comment by mateuszbaran** [comment]
Yes, right, you can't just move the methods that require `RecursiveArrayTools` to Manopt.jl

File: Changelog.md [hunk 3]
```diff
@@ -5,6 +5,54 @@ All notable Changes to the Julia package `Manopt.jl` will be documented in this
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.4.40] – 02/09/2023
```
Comment (mateuszbaran) [inline Changelog.md:8]:
```suggestion
## [0.4.40] – 24/10/2023
```
BTW, I generally prefer YYYY-MM-DD format -- it's less confusing.

Comment (kellertuer) [reply Changelog.md:8]: Oh, too much copy paste. Thanks for noticing. Well, this is the good European dd/mm/yyyy (but I see where Americans can confuse this ;)

File: docs/Project.toml [hunk 4]
```diff
@@ -23,5 +23,5 @@ Random = "9a3f8284-a2c9-5f02-9a11-845980a1fd5c"
 BenchmarkTools = "1.3"
 CondaPkg = "0.2"
 Documenter = "0.27"
```
Comment (mateuszbaran) [inline docs/Project.toml:25]: It would be nice to bumper Documenter too.

Comment (kellertuer) [reply docs/Project.toml:25]: Oh I had thought I had done that already, well spotted, thanks!

File: Changelog.md [hunk 5]
```diff
@@ -5,6 +5,68 @@ All notable Changes to the Julia package `Manopt.jl` will be documented in this
...
+  on a computer, the docs can still be build with the tutorials not being added to the menu
+  such that documenter does not expect them to exist.
+
+### Changes
+
+* Bump dependencies to `ManifoldsBase.jl` 0.15 and `Manifolds.jl` 0.9
+* move the ARC CG subsolver to the main package, since `TangentSpace` is now already
+  available from `ManifoldsBase`.
+
+## [0.4.39] – 02/09/2023
```
Comment (mateuszbaran) [inline Changelog.md:23]:
```suggestion
## [0.4.39] – 09/10/2023
```

File: Changelog.md [hunk 6]
```diff
@@ -5,6 +5,68 @@ All notable Changes to the Julia package `Manopt.jl` will be documented in this
...
+
+### Changes
+
+* avoid allocations when calling `get_jacobian!` within the Levenberg-Marquard Algorithm.
+
+### Fixed
+
+* Fix a lot of typos in the documentation
+
+## [0.4.37] – 02/09/2023
```
Comment (mateuszbaran) [inline Changelog.md:40]:
```suggestion
## [0.4.37] – 28/09/2023
```

File: docs/make.jl [hunk 7]
```diff
@@ -2,6 +2,28 @@
...
 
+if "--help" ∈ ARGS
+    println(
+        """
+docs/make.jl
+
+Render the `Manopt.jl` documenation with optinal arguments
+
+Arguments
+* `--exclude-docs` - exlcude the tutorials from the menu of Documenter,
```
Comment (mateuszbaran) [inline docs/make.jl:13]:
```suggestion
* `--exclude-docs` - exclude the tutorials from the menu of Documenter,
```

File: docs/src/tutorials/HowToDebug.md [hunk 8]
```diff
@@ -74,7 +74,7 @@ There is two more advanced variants that can be used. The first is a tuple of a
 
 We can for example change the way the `:ϵ` is printed by adding a format string
 and use [`DebugCost`](@ref)`()` which is equivalent to using `:Cost`.
-Especially with the format change, the lines are more consistent in length.
+Especially with the format change, the lines are more coniststent in length.
```
Comment (mateuszbaran) [inline docs/src/tutorials/HowToDebug.md:77]: That word was fine 
```suggestion
Especially with the format change, the lines are more consistent in length.
```

**Comment by mateuszbaran** [comment]
I think we could also add changelog to docs like in Manifolds.jl.

File: Changelog.md (hunk 6 above)
Comment (kellertuer) [reply Changelog.md:40]: Sorry for a few copy paste errors.

File: docs/src/tutorials/HowToDebug.md (hunk 8 above)
Comment (kellertuer) [reply docs/src/tutorials/HowToDebug.md:77]: I do not know how this happened.

**Comment by kellertuer** [comment]
Yes, I can do that tomorrow morning.

**Pull request merged**